### ISSUE TITLE
When the alter view name command is executed, "ERROR: String index out of range: -1" is returned, but the alter is actually successful.

### ISSUE THREAD

**Issue by peilinqian** [apache/shardingsphere#21230]
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
we find java version: java8, full_version=1.8.0_342, full_path=/home/peilq_sharding/bisheng-jdk1.8.0_342//bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: dirty-4114e7ee4cbe5923c2b403a3e86d1f23355cadf3
Commit Message: Fix parse exception when execute insert statement with negative value (https://github.com/apache/shardingsphere/pull/21197)
Branch: https://github.com/apache/shardingsphere/commit/4114e7ee4cbe5923c2b403a3e86d1f23355cadf3
Build time: 2022-09-27T16:37:07+0800

### Expected behavior
When the alter view name command is executed, "ATLER VIEW" is returned, the alter is successful.
### Actual behavior
When the alter view name command is executed, "ERROR: String index out of range: -1" is returned, but the alter is actually successful.

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
```
--create rules 
create sharding broadcast table rules (t_broadcast,t_broadcast_view,t_broadcast_view_new);
select pg_sleep(2);
show sharding broadcast table rules;
--create table
create table t_broadcast(id int,c_id int,a int,b char(10));
insert into t_broadcast values (1,1,10,'test11'),(1,2,10,'Test12'),(1,2,null,'test12'),
(2,1,20,'test21'),(2,2,20,'test22'),(2,3,null,'Test22'),
(3,3,null,'test3'),(3,3,null,'test3'),(4,4,40,'test4'),(5,5,50,'');
--create view
show sharding broadcast table rules;
create view t_broadcast_view as select * from t_broadcast;
select * from t_broadcast_view where id=4;
--alter view
alter view t_broadcast_view rename to t_broadcast_view_new;  --error
select pg_sleep(2);
--teardown
drop view if exists t_broadcast_view;
drop view if exists t_broadcast_view_new;
drop table if exists t_broadcast;
drop sharding broadcast table rules;
```

```
test_db=> --alter view
test_db=> alter view t_broadcast_view rename to t_broadcast_view_new;
ERROR:  String index out of range: -1
test_db=> select pg_sleep(2);
 pg_sleep
----------

(1 row)

test_db=> select * from t_broadcast_view_new;
 id | c_id | a  |     b
----+------+----+------------
  1 |    1 | 10 | test11
  1 |    2 | 10 | Test12
  1 |    2 |    | test12
  2 |    1 | 20 | test21
  2 |    2 | 20 | test22
  2 |    3 |    | Test22
  3 |    3 |    | test3
  3 |    3 |    | test3
  4 |    4 | 40 | test4
  5 |    5 | 50 |
(10 rows)
```

**Comment by strongduanmu** [comment]
@peilinqian Thank you for your feedback, I will check this exception.

**Comment by strongduanmu** [comment]
I have test the master branch, this exception is caused by following logic.

```java
/**
 * Schema refresher for alter view statement.
 */
public final class AlterViewStatementSchemaRefresher implements MetaDataRefresher<AlterViewStatement> {
    
    @Override
    public Optional<MetaDataRefreshedEvent> refresh(final ShardingSphereDatabase database, final Collection<String> logicDataSourceNames,
                                                    final String schemaName, final AlterViewStatement sqlStatement, final ConfigurationProperties props) throws SQLException {
        String viewName = sqlStatement.getView().getTableName().getIdentifier().getValue();
        String viewDefinition = sqlStatement.getViewSQL().substring(sqlStatement.getViewSQL().substring(0, sqlStatement.getViewSQL().indexOf(" as ")).length() + 4).trim();
        SchemaAlteredEvent event = new SchemaAlteredEvent(database.getName(), schemaName);
        Optional<SimpleTableSegment> renameView = AlterViewStatementHandler.getRenameView(sqlStatement);
```

**Comment by strongduanmu** [comment]
Following is my test result.

```sql
sharding_db=> show sharding broadcast table rules;
 sharding_broadcast_tables
---------------------------
 t_broadcast
 t_broadcast_view
 t_broadcast_view_new
(3 rows)

sharding_db=> create view t_broadcast_view as select * from t_broadcast;
CREATE VIEW
sharding_db=>
sharding_db=> select * from t_broadcast_view where id=4;
 id | c_id | a  |     b
----+------+----+------------
  4 |    4 | 40 | test4
(1 row)

sharding_db=> alter view t_broadcast_view rename to t_broadcast_view_new;
ALTER VIEW

sharding_db=> select * from t_broadcast_view_new where id=4;
 id | c_id | a  |     b
----+------+----+------------
  4 |    4 | 40 | test4
(1 row)
```

**Comment by wsm12138** [comment]
/bin/java
we find java version: java8, full_version=1.8.0_312, full_path=/bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: a467bbe1668c2c300d3c91236e9e213c565ab7bf
Commit Message: Fix openGauss cursor execution in xa transaction(#21250) (#21257)
Branch: a467bbe1668c2c300d3c91236e9e213c565ab7bf
Build time: 2022-09-29T13:24:09+0800

<img width="857" alt="image" src="https://user-images.githubusercontent.com/86462784/192971138-28564b26-29c9-4671-987d-6564c91aaa2b.png">

<img width="1857" alt="image" src="https://user-images.githubusercontent.com/86462784/192971532-8457536a-aa85-4aef-9ec3-a2495a0b3da1.png">

**Comment by peilinqian** [comment]
we find java version: java8, full_version=1.8.0_282, full_path=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/bin/java
ShardingSphere-5.2.1-SNAPSHOT
Commit ID: dirty-bcde6f374c4a3a025173fbc9f6d0e66ed686a042
Commit Message: Fix fetch forward all error in openGauss(#21421) (#21471)
Branch: bcde6f374c4a3a025173fbc9f6d0e66ed686a042
Build time: 2022-10-11T19:13:56+0800

```
test_db=> --create table
test_db=> create table t_broadcast(id int,c_id int,a int,b char(10));
CREATE TABLE
test_db=> insert into t_broadcast values (1,1,10,'test11'),(1,2,10,'Test12'),(1,2,null,'test12'),
test_db-> (2,1,20,'test21'),(2,2,20,'test22'),(2,3,null,'Test22'),
test_db-> (3,3,null,'test3'),(3,3,null,'test3'),(4,4,40,'test4'),(5,5,50,'');
INSERT 0 10
test_db=> --create view
test_db=> show sharding broadcast table rules;
 sharding_broadcast_tables
---------------------------
 t_broadcast
 t_broadcast_view
 t_broadcast_view_new
(3 rows)

test_db=> create view t_broadcast_view as select * from t_broadcast;
CREATE VIEW
test_db=> select * from t_broadcast_view where id=4;
 id | c_id | a  |     b
----+------+----+------------
  4 |    4 | 40 | test4
(1 row)

test_db=> --alter view
test_db=> alter view t_broadcast_view rename to t_broadcast_view_new;  --error
ALTER VIEW
test_db=> select pg_sleep(2);
 pg_sleep
----------

(1 row)

test_db=> --teardown
test_db=> drop view if exists t_broadcast_view;
DROP VIEW
test_db=> drop view if exists t_broadcast_view_new;
DROP VIEW
test_db=> drop table if exists t_broadcast;
DROP TABLE
test_db=> drop sharding broadcast table rules;
SUCCESS

```
